pages/api/id/[form]/retrieval.ts

```diff
@@ -49,48 +54,66 @@ function connectToDynamo(): DynamoDBDocumentClient {
  */
 async function getFormResponses(
-  req: NextApiRequest,
   res: NextApiResponse,
   formID: string,
   email?: string,
   temporaryToken?: string
 ): Promise<void> {
   try {
     const documentClient = connectToDynamo();
-    //Create form's responses db param
-    //NB: A filter expression is applied after a Query finishes, but before the results are returned.
-    //Therefore, a Query consumes the same amount of read capacity, regardless of whether a filter expression is present.
-    const getItemsDbParams = {
-      TableName: "Vault",
-      IndexName: "retrieved-index",
-      Limit: 10,
-      //Cannot use partitin key or sort key such as formID and Retrieved attribute in keyConditionExpression simultaneously
-      ExpressionAttributeValues: {
-        ":formID": formID,
-        ":retrieved": 0,
-      },
-      KeyConditionExpression: "Retrieved = :retrieved",
-      FilterExpression: "FormID = :formID",
-      //A filter expression cannot contain partition key or sort key attributes.
-      //You need to specify those attributes in the key condition expression, not the filter expression.
-      ProjectionExpression: "FormID,SubmissionID,FormSubmission,Retrieved,SecurityAttribute",
-    };
-
-    const formResponses = await documentClient.send(new QueryCommand(getItemsDbParams));
+
+    const accumulatedResponses: { [key: string]: any }[] = [];
+    let lastEvaluatedKey = null;
+
+    while (lastEvaluatedKey !== undefined) {
+      //Create form's responses db param
+      //NB: A filter expression is applied after a Query finishes, but before the results are returned.
+      //Therefore, a Query consumes the same amount of read capacity, regardless of whether a filter expression is present.
+      const getItemsDbParams: QueryCommandInput = {
+        TableName: "Vault",
+        IndexName: "retrieved-index",
+        // Just being smart and try to get just what we need to get to a maximum of 10 responses
+        Limit: 10 - accumulatedResponses.length,
+        ExclusiveStartKey: lastEvaluatedKey ?? undefined,
+        //Cannot use partitin key or sort key such as formID and Retrieved attribute in keyConditionExpression simultaneously
+        ExpressionAttributeValues: {
+          ":formID": formID,
+          ":retrieved": 0,
+        },
+        KeyConditionExpression: "Retrieved = :retrieved",
+        FilterExpression: "FormID = :formID",
+        //A filter expression cannot contain partition key or sort key attributes.
+        //You need to specify those attributes in the key condition expression, not the filter expression.
+        ProjectionExpression: "FormID,SubmissionID,FormSubmission,Retrieved,SecurityAttribute",
+      };
+
+      const response = await documentClient.send(new QueryCommand(getItemsDbParams));
+
+      if (response.Items) {
+        for (const item of response.Items) accumulatedResponses.push(item);
```

Comment:
Instead of doing this you could use the concat function 

`accumulatedResponses.concat(response.Items)`

Reply:
Absolutely! No idea why I did not do that initially  😕